api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -167,10 +168,23 @@ message FilterChainMatch {
 
 // A filter chain wraps a set of match criteria, an option TLS context, a set of filters, and
 // various other parameters.
-// [#next-free-field: 8]
+// [#next-free-field: 9]
 message FilterChain {
   option (udpa.annotations.versioning).previous_message_type = "envoy.api.v2.listener.FilterChain";
 
+  // The configuration for on-demand filter chain. If this field is not empty in FilterChain message,
+  // a filter chain will be built on-demand.
+  // On-demand filter chains help speedup the warming up of listeners since the building and initialization of
+  // an on-demand filter chain will be postponed to the arrival of new connection requests that require this filter chain.
+  // Filter chains that are not often used should be set as on-demand.
+  message OnDemandConfiguration {
+    // The timeout to wait for filter chain placeholders to complete rebuilding.
+    // If the timeout is reached, the filter chain rebuilding fails. Rebuilder send callbacks with failure signal
+    // and all connections related to this filter chain will not be created.
```

Comment:
```
// The timeout to wait for filter chain placeholders to complete rebuilding.
// If the timeout is reached, the filter chain rebuilding fails and all pending connections will be closed.
```

Reply:
In the case of timeouts are there retries or is it perma fail? It seems like we should retry in the background? Can you clarify that?

Reply:
I suppose that retry rebuilding after timeout makes the timer meaningless. If rebuilding keeps on doing after timeout again and again, why don't we just ignore the timeout.

Reply:
I think there are two options:
1. no timeout -> rebuilding forever until getting result from dependencies.
2. with timeout -> on timeout of rebuilding, we abandon this rebuilding and notify workers to close all connections.

Reply:
I think you want the timeout otherwise you might get into a pretty bad spot with failure retry storms. I think failing all connection after timeout and then retrying on the next new connection is fine. Let's just clarify that in the docs. Thank you!

Reply:
// 1. If this field is set to 0, timeout is disabled. rebuilding will keeps on doing until dependencies are ready.
    // 2. If not specified, a default timeout of 15s is used. If the timeout is reached, the filter chain rebuilding fails. 
    // All connections related to this filter chain will not be created. A new connection will start the rebuilding again.

How about this description?

Reply:
```
// Rebuilding will wait until dependencies are ready, have failed, or this timeout is reached.
// If not specified, a default timeout of 15s is used.
// Upon failure or timeout, all connections related to this filter chain will be closed. Rebuilding will start again on the next new connection.
```

Reply:
So, we will always have a timeout for the rebuilding, right? Or do we need to have option for 'timeout disabled: waiting forever'?

Reply:
Sorry, I forgot that part. Yes, feel free to add if 0 is specified the timeout is disabled.

Reply:
Thanks~ Updated.